File: backends/php/tests/CalculatorTest.php
<?php

declare(strict_types=1);

namespace Aurora\Tests;

use Aurora\Calculator;
use PHPUnit\Framework\TestCase;

final class CalculatorTest extends TestCase
{
    private Calculator $calc;

    protected function setUp(): void
    {
        $this->calc = new Calculator();
    }

    public function testAdd(): void
    {
        $this->assertSame(5.0, $this->calc->add(2, 3));
        $this->assertSame(-8.0, $this->calc->add(-5, -3));
        $this->assertSame(0.0, $this->calc->add(0, 0));
    }

    public function testSubtract(): void
    {
        $this->assertSame(6.0, $this->calc->subtract(10, 4));
        $this->assertSame(-2.0, $this->calc->subtract(-5, -3));
    }

    public function testMultiply(): void
    {
        $this->assertSame(42.0, $this->calc->multiply(6, 7));
        $this->assertSame(12.0, $this->calc->multiply(-3, -4));
        $this->assertSame(0.0, $this->calc->multiply(123, 0));
    }

    public function testDivide(): void
    {
        $this->assertSame(4.0, $this->calc->divide(20, 5));
        $this->assertSame(5.0, $this->calc->divide(-10, -2));
        $this->assertEqualsWithDelta(3.3333, $this->calc->divide(10, 3), 0.0001);
    }

    public function testDivideByZeroThrows(): void
    {
        $this->expectException(\InvalidArgumentException::class);
        $this->expectExceptionMessage('Division par zéro impossible.');
        $this->calc->divide(10, 0);
    }
}

File: backends/java/src/test/java/com/aurora/calc/CalculatorTest.java
package com.aurora.calc;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

class CalculatorTest {

    private final Calculator calc = new Calculator();

    @Test
    void add() {
        assertEquals(5, calc.add(2, 3), 1e-9);
        assertEquals(-8, calc.add(-5, -3), 1e-9);
        assertEquals(0, calc.add(0, 0), 1e-9);
    }

    @Test
    void subtract() {
        assertEquals(6, calc.subtract(10, 4), 1e-9);
        assertEquals(-2, calc.subtract(-5, -3), 1e-9);
    }

    @Test
    void multiply() {
        assertEquals(42, calc.multiply(6, 7), 1e-9);
        assertEquals(12, calc.multiply(-3, -4), 1e-9);
        assertEquals(0, calc.multiply(123, 0), 1e-9);
    }

    @Test
    void divide() {
        assertEquals(4, calc.divide(20, 5), 1e-9);
        assertEquals(5, calc.divide(-10, -2), 1e-9);
        assertEquals(3.3333, calc.divide(10, 3), 1e-4);
    }

    @Test
    void divideByZeroThrows() {
        ArithmeticException ex = assertThrows(ArithmeticException.class, () -> calc.divide(10, 0));
        assertEquals("Division par zéro impossible.", ex.getMessage());
    }
}
